internal/xds/v3/contour.go

```diff
@@ -121,9 +111,6 @@ func (s *contourServer) stream(st grpcStream) error {
 			return done(log, fmt.Errorf("no resource registered for typeURL %q", req.GetTypeUrl()))
```

Comment:
As of this revision, the error log from the `done` func when we reject a request with no type URL will not have the other log fields on it set by the logs in `logDiscoveryRequestDetails`, whether this is an issue or not is an open question

Reply:
IMHO it'd be better not to lose those fields, for debuggability. 

You could extract a helper that returns a FieldLogger for a request with all the appropriate fields set, then explicitly call it here and also inside `logDiscoveryRequestDetails`. That helper would end up getting called twice for each request for the contour server, but that should be fine since the log fields will just get set again, not duplicated.

Reply:
fixed by returning the `*logrus.Entry` so we can use it after populated

Reply:
OK, that works